File: core/src/net/natruid/jungle/views/UnitStatsQuickview.java
package net.natruid.jungle.views;

import com.kotcrab.vis.ui.widget.VisLabel;
import com.kotcrab.vis.ui.widget.VisWindow;

public class UnitStatsQuickview extends AbstractView {
    public final VisLabel hp;
    public final VisLabel intelligence;
    public final VisLabel determination;
    public final VisLabel endurance;
    public final VisLabel awareness;
    public final VisLabel luck;
    public final VisWindow window;

    public UnitStatsQuickview() {
        window = new VisWindow("", "info");
        window.setMovable(false);
        window.defaults().pad(0, 5, 0, 5);
        window.add(new VisLabel("HP", "green-small"));
        hp = new VisLabel("0", "green-small");
        window.add(hp);
        window.row();
        window.add(new VisLabel("I", "green-small"));
        intelligence = new VisLabel("0", "green-small");
        window.add(intelligence);
        window.row();
        window.add(new VisLabel("D", "green-small"));
        determination = new VisLabel("0", "green-small");
        window.add(determination);
        window.row();
        window.add(new VisLabel("E", "green-small"));
        endurance = new VisLabel("0", "green-small");
        window.add(endurance);
        window.row();
        window.add(new VisLabel("A", "green-small"));
        awareness = new VisLabel("0", "green-small");
        window.add(awareness);
        window.row();
        window.add(new VisLabel("L", "green-small"));
        luck = new VisLabel("0", "green-small");
        window.add(luck);
        window.pack();
        window.setWidth(128);
        addActor(window);
    }
}

File: core/src/net/natruid/jungle/views/UnitStatsQuickview.kt
package net.natruid.jungle.views

import com.kotcrab.vis.ui.widget.VisLabel
import com.kotcrab.vis.ui.widget.VisWindow
import ktx.vis.window

class UnitStatsQuickview : AbstractView() {
    lateinit var hp: VisLabel
    lateinit var int: VisLabel
    lateinit var det: VisLabel
    lateinit var end: VisLabel
    lateinit var awa: VisLabel
    lateinit var luc: VisLabel
    val base: VisWindow = window("", "info") {
        isMovable = false
        defaults().pad(0f, 5f, 0f, 5f)
        label("HP", "green-small")
        hp = label("0", "green-small")
        row()
        label("I", "green-small")
        int = label("0", "green-small")
        row()
        label("D", "green-small")
        det = label("0", "green-small")
        row()
        label("E", "green-small")
        end = label("0", "green-small")
        row()
        label("A", "green-small")
        awa = label("0", "green-small")
        row()
        label("L", "green-small")
        luc = label("0", "green-small")
        pack()
        width = 128f
    }

    init {
        addActor(base)
    }
}
